### DISCUSSION TITLE
Poor Performance rendering high res images

### DISCUSSION THREAD

**Discussion by johannes-do** [foxglove/community#665] [Q&A]
I want to render a raw 8 mpx images at ~20 fps. This is not possible with the current foxglove implementation. It has significant lag (around 0.1x real time).

I havent checked what the bottleneck is. The decoding, creating the texture or actually rendering it. So maybe its also three.js being the issue here.

Has anyone looked into this already? Any solutions other than scaling down the image?

I have done some eval on my own visu which does essentially the same as foxglove: 1) Convert the image data to sth tha canvas can display 2) Update the canvas with this data:

![image](https://user-images.githubusercontent.com/15958535/227173842-9ecbdd31-96a0-452a-821e-440e3db846bd.png)

Question is, how can this runtime be reduced 🤔

**Comment by defunctzombie** [comment]
Would you be able to provide some sample data (mcap or bag file) we can use to experience this delay for ourselves?

Also helpful to note with performance numbers is what type of machine you are running on.
